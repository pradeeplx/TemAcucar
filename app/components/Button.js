import React from 'react'
import { Platform, StyleSheet } from 'react-native'
import { fontFactor } from "../helpers"
import ReactNativeButton from 'apsl-react-native-button'
import Colors from "../Colors"

export default Button = ({ textStyle, disabledStyle, secondary, style, children }, ...props) => (
  <ReactNativeButton
    activityIndicatorColor={secondary ? Colors.white : Colors.pink}
    {...props}
    textStyle={[{
      textAlign: 'center',
      color: (secondary ? Colors.white : Colors.pink),
      fontSize: (secondary ? 10 : 12) * fontFactor(),
      fontFamily: (Platform.OS === 'ios' ? 'Avenir' : 'Roboto'),
    }, textStyle]}
    disabledStyle={[{
      opacity: 0.6,
    }, disabledStyle]}
    style={[{
      height: (secondary ? 24 : 36) * fontFactor(),
      alignSelf: 'center',
      borderRadius: (secondary ? 12 : 18) * fontFactor(),
      paddingHorizontal: (secondary ? 20 : 30),
      paddingVertical: 0,
      marginBottom: 0,
      borderWidth: StyleSheet.hairlineWidth,
      borderColor: (secondary ? Colors.white : Colors.gray),
    }, style]}
  >
    {children}
  </ReactNativeButton>
)
