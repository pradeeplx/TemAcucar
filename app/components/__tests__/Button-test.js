import 'react-native'
import React from 'react'
import Button from '../Button'
import renderer from 'react-test-renderer'

const component = renderer.create(
  <Button
    href="#"
    textStyle={{color: 'red'}}
    disabledStyle={{color: 'green'}}
    style={{top: 0}}
  >
    FooBar
  </Button>
);

it('renders correctly', () => {
  expect(component.toJSON()).toMatchSnapshot();
});

it('renders secondary colors', () => {
  const tree = renderer.create(
    <Button {...component.props} secondary />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
