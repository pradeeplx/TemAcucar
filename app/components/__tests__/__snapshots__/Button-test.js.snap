exports[`test renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "borderColor": "#FFFFFF",
      "borderRadius": 18,
      "borderWidth": 0.5,
      "flexDirection": "row",
      "height": 36,
      "justifyContent": "center",
      "marginBottom": 0,
      "opacity": 1,
      "paddingHorizontal": 20,
      "paddingVertical": 0,
      "top": 0,
    }
  }
  testID={undefined}>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
          "fontSize": 18,
          "textAlign": "center",
        },
        Array [
          Object {
            "color": "#FFFFFF",
            "fontFamily": "Avenir",
            "fontSize": 15,
            "textAlign": "center",
          },
          Object {
            "color": "red",
          },
        ],
      ]
    }>
    FooBar
  </Text>
</View>
`;
